apps/web/src/components/Room/Room.tsx

```diff
@@ -33,8 +36,43 @@ const useStyles = makeStyles((theme: Theme) => {
 export default function Room() {
   const classes = useStyles();
   const { isChatWindowOpen } = useChatContext();
-  const { isBackgroundSelectionOpen } = useVideoContext();
-  const { appState } = useAppState();
+  const { isBackgroundSelectionOpen, room } = useVideoContext();
+  const { connect: playerConnect } = usePlayerContext();
+  const { registerUserDocument } = useSyncContext();
+  const { appState, appDispatch } = useAppState();
+  const speakerDepartedRef = useRef(false);
+
```

Comment:
Was thinking about the event name we discussed yesterday `setPreventJoinStreamAsViewer`.  I'm wondering if this would also be confusing because this custom event does get emitted when the speaker clicks on the "Leave and View Event" button, so it might seem a little odd that we are emitting an event that is "preventing" the exact desired behavior 🤔 .